v2/pkg/vm/volume.go

```diff
@@ -184,7 +185,18 @@ func (v *imageVolume) create(ctx context.Context, dataDir, dataPathLastPart stri
 }
 
 func createCoWImageFromBase(ctx context.Context, base, dest string) error {
-	c := well.CommandContext(ctx, "qemu-img", "create", "-f", "qcow2", "-o", "backing_file="+base, dest)
+	var info map[string]interface{}
+	out, err := well.CommandContext(ctx, "qemu-img", "info", "--output=json", base).Output()
+	if err != nil {
+		return err
+	}
+
+	if err := json.Unmarshal(out, &info); err != nil {
+		return err
+	}
+
+	fileFormat := fmt.Sprintf("%v", info["format"])
```

Comment:
Why you use `fmt.Sprintf("%v")`?
If `format` is not a string, `qemu-img create` will fail.
It will also fail if the format is an empty string.

Reply:
You can use the following struct instead of `map[string]interface{}`
```go
var info struct {
	Format string `json:"format"`
}
```

Reply:
I fixed using a struct instead of `map[string]interface{}`.
02a873b2d51723b8b1b8303c1557f4da6e751590